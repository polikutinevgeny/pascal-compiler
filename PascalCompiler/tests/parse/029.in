program read_;
var
   n: integer;

   procedure readln(var n: integer);
   begin
      //some block
   end;

begin
   while n < 0 do
      readln(n);
end.