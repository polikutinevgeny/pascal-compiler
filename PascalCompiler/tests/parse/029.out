program read_
  integer = integer;
  real = real;
  char = char;
  read_ = program read_;
  n = integer;
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
  procedure readln((Var, n, integer))
      n = Var integer;
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
└─while
  ├─Less
  │ ├─n
  │ └─0
  └─Call statement wrapper
    └─readln call
      └─n
